psycopg3/psycopg3/cursor.py

```diff
@@ -516,19 +529,21 @@ def fetchmany(self, size: int = 0) -> Sequence[Sequence[Any]]:
             self._pos, min(self._pos + size, self.pgresult.ntuples)
         )
         self._pos += len(records)
-        return records
+        assert self._make_row is not None
+        return [self._make_row(r) for r in records]
 
-    def fetchall(self) -> Sequence[Sequence[Any]]:
+    def fetchall(self) -> Sequence[Row]:
         """
         Return all the remaining records from the current recordset.
         """
         self._check_result()
         assert self.pgresult
         records = self._tx.load_rows(self._pos, self.pgresult.ntuples)
         self._pos += self.pgresult.ntuples
-        return records
+        assert self._make_row is not None
+        return [self._make_row(r) for r in records]
```

Comment:
`list(map(self_make_row, records))` is maybe more efficient (one lookup less per entry), and entirely avoidable with the default make_row.